lib.hpp

```diff
@@ -1,5 +1,9 @@
 #include <stack>
 #include <string>
+#include <iostream>
+#include <sstream>
+#include <vector>
+#include <iterator>
```

Comment:
most of these headers are only needed in the .cpp file. please include only the needed ones here, the others from there. includers of the .hpp file should not even know that streams are used.